nvflare/job_config/fed_job.py

```diff
@@ -153,10 +159,19 @@ def to(
         self._validate_target(target)
 
         if isinstance(obj, Controller):
-            if target not in self._deploy_map:
-                self._deploy_map[target] = ControllerApp(key_metric=self.key_metric)
-            self._deploy_map[target].add_controller(obj, id)
+            if target != "server":  # add client-side controllers as components
+                if target not in self._deploy_map:
+                    raise ValueError(f"{target} doesn't have an `Executor`. Deploy one first before adding components!")
+                self._deploy_map[target].add_component(obj, id)
```

Comment:
suggest enhance the error message to:  `raise ValueError(f"{target} doesn't have an `Executor`. Deploy one first before adding client-side controllers!")`
to be more precise.

So users need to first deploy the Executor before they add the Client side controller?